### PR TITLE
[DUOS-1968][risk=no] Dataset Submission Schema

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1968

Invision mocks for reference: https://projects.invisionapp.com/d/main?origin=v7#/console/22055739/468285703/preview

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/data-submitter-schema_v1.json [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+      "label": "NIH Institution/Center for Submission",
+      "description": "NIH Institution/Center for Submission",
+      "enum": ["NCI", "NEI", "NHLBI", "NHGRI", "NIA", "NIAAA", "NIAID", "NIAMS", "NIBIB", "NICHD", "NIDCD", "NIDCR", "NIDDK", "NIDA", "NIEHS", "NIGMS", "NIMH", "NIMHD", "NINDS", "NINR", "NLM", "CC", "CIT", "CSR", "FIC", "NCATS", "NCCIH"]
+    },
+    "nihGenomicProgramAdministratorName": {
+      "type": "string",
+      "label": "NIH Genomic Program Administrator Name",
+      "description": "NIH Genomic Program Administrator Name"
+    },
+    "multiCenterStudy": {
```
Comment (connorlbark) [inline src/main/resources/data-submitter-schema_v1.json:158]: Not sure where all of the fields come from, but if `multiCenterStudy` was true, does this imply there are multiple NIH centers? If so, where would those be denoted?

**Review by connorlbark — APPROVED** [review]
I'm sure we will find things to change once we start using the schema but at a first pass it looks solid to me.

File: src/main/resources/data-submitter-schema_v1.json (hunk 1 above)
Comment (rushtong) [reply src/main/resources/data-submitter-schema_v1.json:158]: Those are defined in the next property, `collaboratingSites`. It's supposed to be an array of (free text) sites, but I'm not sure how the UI will be able to supply that at the moment.

**Review by shaemarks — APPROVED** [review]
LGTM

File: src/main/resources/data-submitter-schema_v1.json [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+          "description": "# of Participants"
+        }
+      }
+    },
+    "consentGroup": {
+      "required": ["consentGroupName"],
+      "oneOf": [
+        {"required":  ["generalResearchUse"]},
+        {"required":  ["hmb"]},
+        {"required":  ["diseaseSpecificUse"]}
```
Comment (connorlbark) [inline src/main/resources/data-submitter-schema_v1.json:286]: does this also need to have `poa`/`otherPrimary` as an option in the `oneOf`?

File: src/main/resources/data-submitter-schema_v1.json [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+          "description": "Functional Equivalence"
+        },
+        "numberOfParticipants": {
+          "type": "number",
+          "description": "# of Participants"
+        }
+      }
+    },
+    "consentGroup": {
+      "required": ["consentGroupName"],
```
Comment (connorlbark) [inline src/main/resources/data-submitter-schema_v1.json:282]: should URL be required?

File: src/main/resources/data-submitter-schema_v1.json (hunk 2 above)
Comment (rushtong) [reply src/main/resources/data-submitter-schema_v1.json:286]: `poa` and `otherPrimary` are completely optional. There has to be at most one of these three primary use restrictions, and all others are secondary and optional.

File: src/main/resources/data-submitter-schema_v1.json (hunk 3 above)
Comment (rushtong) [reply src/main/resources/data-submitter-schema_v1.json:282]: Good question - will follow up with @solideoglori

Comment (rushtong) [reply src/main/resources/data-submitter-schema_v1.json:282]: Follow on to this - we are definitely going to revise this due to other teams' needs. Will merge for now and we can address any new requirements in a follow-on story.

**Pull request merged**